sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -109,10 +109,8 @@ trait CheckAnalysis extends PredicateHelper {
             failAnalysis(s"Distinct window functions are not supported: $w")
 
           case w @ WindowExpression(_: OffsetWindowFunction, WindowSpecDefinition(_, order,
-               SpecifiedWindowFrame(frame,
-                 FrameBoundary(l),
-                 FrameBoundary(h))))
-             if order.isEmpty || frame != RowFrame || l != h =>
+               frame: SpecifiedWindowFrame))
```

Comment:
Nit: Cutting in the middle looks weird. Please keep them in the same line.
`WindowSpecDefinition(_, order, frame: SpecifiedWindowFrame))`